Generals/Code/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp

```diff
@@ -2734,6 +2734,13 @@ GameMessageDisposition CommandTranslator::translateGameMessage(const GameMessage
 
 		}	// end select previous worker
 
+		//-----------------------------------------------------------------------------------------
+		case GameMessage::MSG_META_SELECT_NEXT_IDLE_WORKER:
+			if (TheInGameUI)
+				TheInGameUI->selectNextIdleWorker();
+			disp = DESTROY_MESSAGE;
```

Comment:
```cpp
if (TheInGameUI)
{
  TheInGameUI->selectNextIdleWorker();
  disp = DESTROY_MESSAGE;
}
```

Or maybe even

```cpp
if (TheInGameUI)
{
  if (TheInGameUI->selectNextIdleWorker())
  {
    disp = DESTROY_MESSAGE;
  }
}
```

It would require returning a bool in that function